include/core/truthTable/truth_table.hpp

```diff
@@ -280,6 +329,21 @@ namespace syrec {
             cubeMap.insert(std::move(nh));
         }
 
+        static auto equal(TruthTable& tt1, TruthTable& tt2, const bool equalityUpToDontCare = true) -> bool {
+            if (!equalityUpToDontCare) {
+                return (tt1 == tt2);
+            }
+            for (auto const& [input, output]: tt1) {
+                if (auto it = tt2.find(input); it == tt2.end()) {
+                    continue;
+                }
```

Comment:
This is not symmetric. As a consequence `equal(a, b)` is not necessarily equal to `equal(b, a)`. This is unintuitive and unnecessary.